pkg/edgeview/src/crypto.go

```diff
@@ -51,7 +51,9 @@ func addEnvelopeAndWriteWss(msg []byte, isText bool) error {
 	} else {
 		msgType = websocket.BinaryMessage
 	}
+	wssWrMutex.Lock()
```

Comment:
Can't the write on the websocket block for minutes? Until TCP times out.
What happens to other code if they try to acquire the mutex in such a case?
(Typically mutexes are held for very short times.)

Reply:
don't think so. we are basically 'packet' based, we get a tcp 'packet' from one side, then write to the other side of the channel. we use websocket to relay those 'packet' for tcp, we only read a max number of bytes at a time when writing onto the socket.

Reply:
I don't understand why you need to hold a mutex across the calls to websocketConn.WriteMessage(). Can you explain why and what fails when you don't? 
Seems like this can result in potential deadlocks since in worst case you're blocking the caller until TCP has timed out

Reply:
@eriknordmark for websocket connection, only one can write at this at anytime, otherwise will trigger the panic like this:
  panic: concurrent write to websocket connection
different channel writers need to get the mutex in order to write this.
the write only for max of 8k bytes, should be very fast.

Reply:
OK, then it makes sense to add a comment where the mutex is declared to state that it unsures that we do not have concurrent calls to websocketConn.WriteMessage().

Note that the hold time can be very long. If the receiver stops consuming data from tcp e.g., due to packets being dropped by the network or the receiving process being very flow, then at some point in time I assume that the WriteMessage will block due to flow control. If the packets are dropped then TCP will time out (after a few minutes) but if the receiving process is very slow but TCP is responding with Acks, then a WriteMessage can block forever AFAUI.

But in this case that is probably ok since nothing else but WriteMessage is protected  by the mutex. Just need the comment to explain this to the next person to read the code.

Reply:
comments added. to see if this is ok.